src/crypto/OlmDevice.js

```diff
@@ -270,9 +268,13 @@ OlmDevice.prototype._getUtility = function(func) {
  * @return {Promise<string>} base64-encoded signature
  */
 OlmDevice.prototype.sign = async function(message) {
-    return await this._getAccount(function(account) {
-        return account.sign(message);
+    let result;
```

Comment:
again, can't we pass the result up through the callbacks to save us having to fiddle with a local result?

(perhaps it actually ends up being more confusing?)

Reply:
we can, but it could now be quite a few callbacks, and you could be doing more than one get in the transaction, at which point it's not clear which one the transaction promise would return, so I think I'd vote for just making you assign it to a captured variable.

Reply:
the transaction promise would return whatever is returned by the `func` passed to `doTxn`.

I don't think I agree with you, but I don't feel super strongly about it.